test: add integration tests

akash2237778 opened the pull request:
## Description
Added Integration tests that test the functionality of each endpoint.

Fixes #39

akash2237778 committed:
test: add integeration tests

uniqueg commented on tests/integration_tests.py:
```diff
@@ -0,0 +1,174 @@
+base_url = "http://localhost:8080/ga4gh/drs/v1"
+data_objects_path = "data_objects.json"
+headers = {
+    'accept': 'application/json',
```
I think it should be `Accept` with a capital `A`: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept

uniqueg commented on tests/integration_tests.py:
```diff
@@ -0,0 +1,174 @@
+    'accept': 'application/json',
+    'Content-Type': 'application/json'
+}
+payload_objects = json.dumps({
```
Can some object from "data_objects.json" perhaps be reused here instead of hardcoding another 50 lines of payload in here?

uniqueg commented on tests/integration_tests.py:
```diff
@@ -0,0 +1,174 @@
+        "alias1"
+    ]
+})
+payload_service_info = json.dumps({
```
Can the payload JSON perhaps be externalized so that it can be easily reused and keep this module cleaner? See here, e.g.: https://github.com/elixir-cloud-aai/trs-filer/blob/dev/tests/mock_data.py

uniqueg commented on tests/integration_tests.py:
```diff
@@ -0,0 +1,174 @@
+    endpoint = "/objects/" + object_id
+    response = requests.request("DELETE", base_url + endpoint, headers=headers)
+    assert response.status_code == 200
+    assert response.text.replace("\"", "").replace("\n", "") == object_id
```
Would be good to add test cases for some errors as well, especially 404 and 400. Eventually we should cover 401 and 403 as well, but we can do that once we have a better idea of how controlled access management will look like

uniqueg reviewed (changes requested):
This looks good - thanks a lot! Some minor refactoring to clean this up a little and would be good to add some tests for `400` (for all endpoints) and `404` (for `GET` endpoints only) responses. I'm also wondering why Travis CI wasn't triggered.... Any idea?

uniqueg commented on .travis.yml:
```diff
@@ -7,7 +7,10 @@ language: python
 # PRs from forks and pushes to existing PRs from forks
 if: (type == push) OR (type == pull_request AND fork == true)
-
+branches:
```
Oh, this is why it didn't trigger a Travis run. Why did you add that? I think we would like to run tests always, for each PR. If you would like for specific tests (like integration tests), you should make use of different stages or a job matrix, then introduce conditions for branches separately for each stage/job. It should also be possible to use conditions on the exact commands, but the syntax is not great and it can quickly become unreadable for more complicated files. In any case, I don't see a good reason why we should not run the Travis CI for all PRs as well, the tests should be fast enough...

uniqueg replied:
(so I would just remove this block again unless there is a convincing reason to keep it)

akash2237778 committed:
test: added integration tests

Added integration tests for responses  and  & created mock_data.py to avoid hard coding payload at integration_tests.py

akash2237778 committed:
test: added integration tests

Minor fix: mock_data

akash2237778 commented:
Made required changes. But still not sure about tests for response `400`, although added for `POST /objects` and `POST /service-info`. Some endpoints never return response 400 e.g. [server.py](https://github.com/elixir-cloud-aai/drs-filer/blob/122c34d5b71317c7b4b809287207677d399cd3e8/drs_filer/ga4gh/drs/server.py#L70). 

I think there is a need to add tests for response `404` at `DELETE /objects/{object_id}/access/{access_id}` and `DELETE /objects/{object_id}` endpoints too.
